zarr/storage.py

```diff
@@ -1779,8 +1779,14 @@ def __setitem__(self, key, value):
             self.zf.writestr(keyinfo, value)
 
     def __delitem__(self, key):
-        raise NotImplementedError
-
+        # raise NotImplementedError
+        if key not in self:
+            raise KeyError("Can not delete a nonexistent key")
+        
+        self[key] = b""
+
+    def pop(self, key):
+        del self
```

Comment:
This is [implemented in the `MutableMapping` base class]( https://github.com/python/cpython/blob/a834e2d8e1230c17193c19b425e83e0bf736179e/Lib/_collections_abc.py#L933-L945 ). So we can skip implementing it here

```suggestion
```